In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
data = [
    100, 150, 50, 100, 130, 120, 100, 85, 70, 150, 150, 120, 50, 100, 100, 140, 90, 150, 50, 90, 120, 
    100, 110, 75, 65
]

df = pd.DataFrame (data, columns=['TransactionAmount'])
df

,TransactionAmount
0,100
1,150
2,50
3,100
4,130
5,120
6,100
7,85
8,70
9,150


In [3]:
# Measures of Central Tendency

mean_value = df['TransactionAmount'].mean()
median_value = df['TransactionAmount'].median()
mode_value = df['TransactionAmount'].mode()

mean_value, median_value, mode_value


(np.float64(102.6),
 100.0,
 0    100
 Name: TransactionAmount, dtype: int64)

Mean = 102.6
Median = 100.0
Mode = 100


In [4]:
# Measures of Variability

range_value = df['TransactionAmount'].max() - df['TransactionAmount'].min()
variance_value = df['TransactionAmount'].var()
std_value = df['TransactionAmount'].std()
min_value = df['TransactionAmount'].min()
max_value = df['TransactionAmount'].max()
Q1 = df ['TransactionAmount'].quantile(0.25)
Q2 = df ['TransactionAmount'].quantile(0.50)
Q3 = df ['TransactionAmount'].quantile(0.75)
range_value, variance_value, std_value, min_value, max_value, Q1, Q2, Q3

(100,
 1012.7499999999997,
 31.82373328193912,
 50,
 150,
 np.float64(85.0),
 np.float64(100.0),
 np.float64(120.0))

Range = 100
Variance = 1012.7499999999997
Standard deviation = 31.82373328193912
min = 50
max = 150
q1 = 85
q2 = 100
q3 = 120

While the typical transaction value remains around 100, there is substantial variation in sales performance.

In [6]:
# Defining H0 and H1
# H0 : μ = 100
# H1 : μ > 100
# One Sample T-test if a = 0.05

from scipy import stats
t_stat, p_value = stats.ttest_1samp(df['TransactionAmount'], 100)
t_stat, p_value

(np.float64(0.4085001556802841), np.float64(0.6865284813438117))

In [7]:
p_value_one_tailed = p_value / 2
p_value_one_tailed


np.float64(0.34326424067190586)

p-value = 0.34326424067190586 > 0.05
fail to reject H0


In [8]:
# T-statistics and Critical region
from scipy.stats import t

In [9]:
alpha = 0.05
df = 24

t_critical = t.ppf(1 - alpha, df)
t_critical

np.float64(1.7108820799094275)

0.4085001556802841 < 1.7108820799094275
fail to reject H0
The increase is not statistically significant at the 5% level. 